src/main/java/org/jsoup/select/HTMLTableExtractor.java

```diff
@@ -0,0 +1,243 @@
+        return parseHtmlTable(doc);
+    }
+
+    /**
+     * Extracts HTML tables from a given HTML string.
+     * 
+     * @param htmlString The HTML string from which to extract HTML tables
+     * 
+     * @return A list of HTMLTable objects representing the extracted tables
+     * 
+     * @throws IOException If an I/O error occurs while parsing the HTML
+     */
+    public List<HTMLTable> extractTableFromHTML(String htmlString) throws IOException {
+        // Validate the htmlString
+        Validate.notEmpty(htmlString, "HTML string must not be empty");
+        
+        // parse the HTML string and retrieve the document
+        Document doc = Jsoup.parse(htmlString);
+        return parseHtmlTable(doc);
+    }
+
+    /**
+     * Extracts HTML tables from a given HTML file path.
+     * 
+     * @param filepath The file path of the HTML file from which to extract HTML tables
+     * 
+     * @return A list of HTMLTable objects representing the extracted tables
+     * 
+     * @throws IOException If an I/O error occurs while reading the HTML file or parsing the HTML
+     */
+    public List<HTMLTable> extractTableFromHTMLFile(String filepath) throws IOException {
+        // Validate the filepath
+        Validate.notEmpty(filepath, "File path must not be empty");
+        
+        // create a File object from the file path
+        File inputFile = new File(filepath);
+        
+        // Check if the file exists and is a regular file
+        Validate.isTrue(inputFile.exists() && inputFile.isFile(), "File does not exist or is not a regular file: " + filepath);
+
+        // parse the HTML file and retrieve the document
+        Document doc = Jsoup.parse(inputFile, "UTF-8");
+        return parseHtmlTable(doc);
+    }
+    
+    /**
+     * Extracts HTML tables from a given HTML file.
+     * 
+     * @param inputFile The HTML file from which to extract HTML tables
+     * 
+     * @return A list of HTMLTable objects representing the extracted tables
+     * 
+     * @throws IOException If an I/O error occurs while reading the HTML file or parsing the HTML
+     */
+    public List<HTMLTable> extractTableFromHTMLFile(File inputFile) throws IOException {
+        // Validate the inputFile
+        Validate.notNull(inputFile, "Input file must not be null");
+
+        // Check if the file exists and is a regular file
+        Validate.isTrue(inputFile.exists() && inputFile.isFile(), "File does not exist or is not a regular file: " + inputFile.getAbsolutePath());
+
+        // parse the HTML file and retrieve the document
+        Document doc = Jsoup.parse(inputFile, "UTF-8");
+        return parseHtmlTable(doc);
+    }
+
+    /**
+     * 
+     * Parses HTML tables from a Jsoup Document object.
+     * 
+     * @param doc The Jsoup Document object representing the HTML
+     * 
+     * @return A list of HTMLTable objects representing the parsed tables
+     */
+    public List<HTMLTable> parseHtmlTable(Document doc) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.